x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ldap/support/LdapMetadataResolver.java

```diff
@@ -76,8 +78,13 @@ private Attribute findAttribute(Collection<Attribute> attributes, String name) {
                                 attr -> attr.getName(),
                                 attr -> {
                                     final String[] values = attr.getValues();
+                                    if(attr.getName().equals(TOKEN_GROUPS)) {
+                                        return Arrays.stream(attr.getValueByteArrays())
+                                            .map((sidBytes) -> convertToString(sidBytes))
+                                            .collect(Collectors.toList());
```

Comment:
How about we return a `String` here instead of a list with one item if tokenGroups contains only 1 value, similar to what we do for all other attributes ( see below ) ?

Reply:
Done